src/esm_runscripts/filelists.py

```diff
@@ -999,59 +1012,72 @@ def copy_files(config, filetypes, source, target):
                     dest_dir = os.path.dirname(file_target)
                     file_source = resolve_symlinks(config, file_source)
                     if not os.path.isdir(file_source):
-                        try:
-                            if not os.path.isdir(dest_dir):
-                                # MA: ``os.makedirs`` creates the specified directory
-                                # and the parent directories if the last don't exist
-                                # (same as with ``mkdir -p <directory>>``)
-                                os.makedirs(dest_dir)
-                            if not os.path.isfile(file_source):
-                                logger.warning(
-                                    f"File not found: {file_source}",
-                                )
-                                helpers.print_datetime(config)
-                                missing_files.update({file_target: file_source})
-                                continue
-                            if os.path.isfile(file_target) and filecmp.cmp(
-                                file_source, file_target
-                            ):
-                                logger.debug(
-                                    f"Source and target file are identical, skipping {file_source}",
-                                )
-                                helpers.print_datetime(config)
-                                continue
-                            files_to_be_moved.append({
+                        if not os.path.isdir(dest_dir):
+                            # MA: ``os.makedirs`` creates the specified directory
+                            # and the parent directories if the last don't exist
+                            # (same as with ``mkdir -p <directory>>``)
+                            os.makedirs(dest_dir)
+                        if not os.path.isfile(file_source):
+                            logger.warning(f"File not found: {file_source}")
+                            helpers.print_datetime(config)
+                            missing_files.update({file_target: file_source})
+                            continue
+                        if os.path.isfile(file_target) and filecmp.cmp(
+                            file_source, file_target
+                        ):
+                            logger.debug(
+                                f"Source and target file are identical, skipping {file_source}",
+                            )
+                            helpers.print_datetime(config)
+                            continue
+                        files_to_be_moved.append(
+                            {
```

Comment:
I don't understand this change. I assume it is needed for parallel copying to work correctly?

Reply:
alreaday discussed in person